src/rin2016/module8/rubber_array/Array.java

```diff
@@ -0,0 +1,241 @@
+package rin2016.module8.rubber_array;
+
+public class Array {
+	private int[] array;
+	private int size;
+	
+	public final static int SEARCH_ERROR = -1;
+	private final int DEFAULT_SIZE = 5;
+	
+	public Array() {
+		create(DEFAULT_SIZE);
+		
+	}
+	
+	public Array(final int size) {
+		create(size);
+	}
+	
+	private void create(final int size) {
+		if (size >= 0) {
+			this.array = new int[size];
+			this.size = 0;
+		} else {
+			this.array = new int[DEFAULT_SIZE];
+			this.size = 0;
+		}
+	}
+	
+	public int getElement(final int index) {
+		if(index <= size && index >= 0) {
+			return array[index];
+		}
+		return Integer.MIN_VALUE;
+	}
+	
+	private int[] getArray() {
+		int[] result = new int [size];
+		copy(result, array, size);
+		return result;
+	}
+	
+	public int size() {
+		return this.size;
+	}
+	
+	/*
+	 * Добавляем элемент в конец массива
+	 */
+	public void add(final int element) {
+		if (size == 0) {
+			this.array[size++] = element;

```

Comment:
It is too risky. You do not have bounds check here. Are you sure you would not have an array with length = 0?